mlflow/models/evaluation/default_evaluator.py

```diff
@@ -0,0 +1,487 @@
+                    self.predict_fn, renamed_X, feature_names=truncated_feature_names
+                )
+                shap_values = explainer(renamed_X, sample_rows)
+            else:
+                explainer = shap.Explainer(
+                    self.predict_fn, sampled_X, feature_names=truncated_feature_names, algorithm=algorithm
+                )
+                shap_values = explainer(sampled_X)
+        else:
+            if self.raw_model and not is_multinomial_classifier:
+                # For mulitnomial classifier, shap.Explainer may choose Tree/Linear explainer for
+                # raw model, this case shap plot doesn't support it well, so exclude the
+                # multinomial_classifier case here.
+                explainer = shap.Explainer(self.raw_model, sampled_X, feature_names=truncated_feature_names)
+                shap_values = explainer(sampled_X)
+            else:
+                # fallback to default explainer
+                explainer = shap.Explainer(
+                    self.predict_fn, sampled_X, feature_names=truncated_feature_names
+                )
+                shap_values = explainer(sampled_X)
+
+        _logger.info(f'Shap explainer {explainer.__class__.__name__} is used.')
+
+        try:
+            mlflow.shap.log_explainer(
+               explainer,
+               artifact_path=_gen_log_key('explainer', self.dataset_name)
+            )
+        except Exception as e:
+            # TODO: The explainer saver is buggy, if `get_underlying_model_flavor` return "unknown",
+            #   then fallback to shap explainer saver, and shap explainer will call `model.save`
+            #   for sklearn model, there is no `.save` method, so error will happen.
+            _logger.warning(f'Log explainer failed. Reason: {str(e)}')
+
+        def plot_beeswarm():
+            pyplot.subplots_adjust(bottom=0.2, left=0.4)
+            shap.plots.beeswarm(shap_values, show=False)
+
+        self._log_image_artifact(
+            plot_beeswarm, "shap_beeswarm_plot",
+        )
+
+        def plot_summary():
+            pyplot.subplots_adjust(bottom=0.2, left=0.4)
+            shap.summary_plot(shap_values, show=False)
+
+        self._log_image_artifact(
+            plot_summary, "shap_summary_plot",
+        )
+
+        def plot_feature_importance():
+            pyplot.subplots_adjust(bottom=0.2, left=0.4)
+            shap.plots.bar(shap_values, show=False)
+
+        self._log_image_artifact(
+            plot_feature_importance,
+            "shap_feature_importance_plot",
+        )
+
+    def _evaluate_per_class(self, positive_class, y, y_pred, y_proba):
+        """
+        if positive_class is an interger, generate metrics and artifacts on this class vs. rest,
+         and the y/y_pred/y_proba must be sum up to a binary "is class" and "is not class"
+        if positive_class is None, generate metrics and artifacts on binary y/y_pred/y_proba
```

Comment:
```suggestion
        If positive_class is None, generate metrics and artifacts on binary y/y_pred/y_proba.
```

If `positive_class` is `None`, can we represent it as a keyword argument with default value `None`?